### DISCUSSION TITLE
Repository Pattern - A controversy explained

### DISCUSSION THREAD

**Discussion by giscus** [linkdotnet/Blog.Discussions#45] [General]
# Repository Pattern - A controversy explained

In this blog post, we'll dive into the ins and outs of the repository pattern and examine both its benefits and its potential drawbacks. We will start from the very basic to some more advanced use cases. So let's dive right into it.

https://steven-giesel.com/blogPost/9fa7fe83-3ede-4ecb-ab27-4012b1333c0e

**Comment by hschmale** [comment]
In IBlogRepository<T> the type 'BlogPost' should be replaced by 'T'.

**Reply by linkdotnet** [reply]
Yes it should - thanks. Updated and corrected the blog post

**Comment by HanaaMohammed96** [comment]
❤❤👌

**Comment by ChristianLavigne** [comment]
I always, and only, use the repository pattern, I have been doing so for a very long time. When done properly, it provides: Clear easy to understand interfaces, encapsulate the complexity of coding for the database and reduces coupling with the storage technologies.

I do have doubts about the generic version. For the interfaces I would rather have the version in "The basics" section. While the generic version offers common functionality, it makes it hard to extend. How to you add things like GetByName, GetTitleContains... ? You can tailor the functionality with "The basics" version, not as simple in the generic version.

**Reply by linkdotnet** [reply]
These are absolutely valid points especially the latter section about the generic version.

You can achieve this with expressions but then again you might make your code very complex.
There are many of those implementations in the wild.
